packages/toolbox-core/README.md

```diff
@@ -98,11 +133,61 @@ Toolbox service.
 from toolbox_core import ToolboxClient
 
 # Replace with your Toolbox service's URL
-toolbox = ToolboxClient("http://127.0.0.1:5000")
+with ToolboxClient("http://127.0.0.1:5000") as toolbox:
 ```
 
 All interactions for loading and invoking tools happen through this client.
 
+### Client Lifecycle Management
+
+The `ToolboxClient` uses an `aiohttp.ClientSession` for its asynchronous network requests. Proper management of this session is key to avoiding resource leaks and warnings.
+
+#### Scenario 1: `ToolboxClient` Manages the Session (Default & Common)
+
+If you initialize `ToolboxClient` without passing a `session` argument, it creates and manages its own `aiohttp.ClientSession`. In this case, you must ensure the `ToolboxClient` instance is closed to also close the underlying session.
+
+##### Recommended Method: `async with`
+```py
+async def main():
+    async with ToolboxClient(toolbox_url) as client:
+        # ... use client ...
+```
+##### Alternative: Explicit `await client.close()`
+```py
+async def main():
+    client = ToolboxClient(toolbox_url)
+    try:
+        # ... use client ...
+    finally:
+        await client.close()
+```
```

Comment:
I don't think we should take an opinionated take on which of these is preferable. If we are doing something like an application w/ LangGraph, the client is a long lived variable and we will probably not use either of these approaches and will just call client.close() on shutdown.